tendermint/src/rpc/id.rs

```diff
@@ -3,9 +3,17 @@
 use getrandom::getrandom;
 use serde::{Deserialize, Serialize};
 
+#[derive(Clone, Debug, Deserialize, Serialize, Eq, PartialEq, Ord, PartialOrd)]
+#[serde(untagged)]
+pub enum IdType {
+    Num(i64),
+    Str(String),
+    None,
+}
+
 /// JSONRPC ID: request-specific identifier
 #[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
-pub struct Id(String);
+pub struct Id(IdType);
```

Comment:
Instead of the wrapper, couldn't this directly be `pub enum Id`? My understanding is that serde can handle these (didn't try though).

Reply:
Maybe. That's a good idea. I'll try it.

Reply:
Yup, you're right, it works. I've submitted it into the PR.

Reply:
Nice one :-)